### PR TITLE
[pytorch] [build] PyTorch 1.9.1 Release Training

### PR DESCRIPTION
**Pull request by ydaiming**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron"/"graviton">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/buildspec-1-9-0.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
+account_id: &ACCOUNT_ID <set-$ACCOUNT_ID-in-environment>
```
Comment (jeet4320) [inline pytorch/buildspec-1-9-0.yml:1]: you can delete this file, it is not needed

File: test/sagemaker_tests/pytorch/training/integration/local/test_single_machine_training.py [hunk 2]
```diff
@@ -42,8 +42,8 @@ def test_mnist(docker_image, processor, instance_type, sagemaker_local_session,
 @pytest.mark.skip_py2_containers
 def test_fastai_mnist(docker_image, instance_type, py_version, sagemaker_local_session, tmpdir):
     _, image_framework_version = get_framework_and_version_from_tag(docker_image)
-    if Version(image_framework_version) == Version("1.9"):
-        pytest.skip("Fast AI is not supported on PT 1.9 DLC")
+    if Version(image_framework_version) == Version("1.9") or Version(image_framework_version) == Version("1.9.1"):
```
Comment (jeet4320) [inline test/sagemaker_tests/pytorch/training/integration/local/test_single_machine_training.py:45]: lets add `in` to skip it for all pt1.9. Will it not be supported for further versions as well

Version(image_framework_version) in Version("1.9")

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py [hunk 3]
```diff
@@ -114,6 +114,9 @@ def test_dist_operations_fastai_gpu(framework_version, ecr_image, sagemaker_regi
     if Version(image_framework_version) == Version("1.9"):
         pytest.skip("Fast ai is not supported on PyTorch v1.9 ")
 
+    if Version(image_framework_version) == Version("1.9.1"):
```
Comment (jeet4320) [inline test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py:117]: same as above, lets add more general filter than checking with pinned version

**Review by jeet4320 — CHANGES_REQUESTED** [review]

File: pytorch/buildspec-1-9-0.yml (hunk 1 above)
Comment (waytrue17) [reply pytorch/buildspec-1-9-0.yml]: Done

File: test/sagemaker_tests/pytorch/training/integration/local/test_single_machine_training.py (hunk 2 above)
Comment (waytrue17) [reply test/sagemaker_tests/pytorch/training/integration/local/test_single_machine_training.py]: Done

File: test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py (hunk 3 above)
Comment (waytrue17) [reply test/sagemaker_tests/pytorch/training/integration/sagemaker/test_distributed_operations.py]: Done

File: test/sagemaker_tests/pytorch/training/integration/local/test_single_machine_training.py (hunk 2 above)
Comment (waytrue17) [reply test/sagemaker_tests/pytorch/training/integration/local/test_single_machine_training.py]: @jeet4320 Seems `Version(image_framework_version) in Version("1.9")` is not working, could you take a look?

**Comment by waytrue17** [comment]
> PR looks good, please revert other changes

Reverted

**Comment by ydaiming** [comment]
Deleted the `pytorch/buildspec-1-9-0.yml` file, since only the latest patch release `1.9.1` would be supported. The sanity check failed because this PR only contains dockerfiles for 1.9.1 training, and fails the check for 1.9.1 inference.
